tests/unit_tests/ode_solver_unit_test/explicit_ode_solver.cpp

```diff
@@ -0,0 +1,217 @@
+            //refines spatial discretization 5 times and writes a convergence table to file
+            //Can be used to verify spatial order of the explicit ODE solver
+            //Note: 2D is implemented but is not included in CMakeLists.txt (untested)
+
+
+            typename dealii::Triangulation<dim>::MeshSmoothing(
+                dealii::Triangulation<dim>::smoothing_on_refinement |
+                dealii::Triangulation<dim>::smoothing_on_coarsening));
+
+    int testfail = 0;
+
+    double left = 0.0;
+    double right = 2.0;
+    const bool colorize = true;
+    int n_refinements = 5;
+
+    //generating grid
+    dealii::GridGenerator::hyper_cube(*grid, left, right, colorize);
+    //setting periodic BC 
+    std::vector<dealii::GridTools::PeriodicFacePair<typename Triangulation::cell_iterator> > matched_pairs;
+    if (dim == 2){
+        dealii::GridTools::collect_periodic_faces(*grid,0,1,0,matched_pairs);
+        dealii::GridTools::collect_periodic_faces(*grid,2,3,1,matched_pairs);
+    }
+    else if (dim == 1){
+        dealii::GridTools::collect_periodic_faces(*grid,
+                0, //left
+                1, //right
+                0, //periodic in x-direction
+                matched_pairs);
+    }
+
+    grid->add_periodicity(matched_pairs);
+    grid->refine_global(n_refinements);
+
+    //default parameters
+    dealii::ParameterHandler parameter_handler;
+    PHiLiP::Parameters::AllParameters::declare_parameters (parameter_handler); // default fills options
+    PHiLiP::Parameters::AllParameters all_parameters;
+    all_parameters.parse_parameters (parameter_handler); // copies stuff from parameter_handler into all_parameters
+
+    //parameters consistent with MPI_2D_ADVECTION_EXPLICIT_PERIODIC_LONG test
+    all_parameters.ode_solver_param.ode_solver_type = PHiLiP::Parameters::ODESolverParam::ODESolverEnum::explicit_solver;
+    all_parameters.ode_solver_param.nonlinear_max_iterations = 500;
+    all_parameters.ode_solver_param.print_iteration_modulo = 100;
+    all_parameters.ode_solver_param.ode_output = PHiLiP::Parameters::OutputEnum::quiet; 
+    
+    int rk_order = 3;
+    double expected_order = -double(rk_order);
+    double order_tolerance = 0.1; 
+    all_parameters.ode_solver_param.runge_kutta_order = rk_order;
+
+    double adv_speed_x = 1.0, adv_speed_y = 0.0;
+    all_parameters.manufactured_convergence_study_param.manufactured_solution_param.advection_vector[0] = adv_speed_x; //x-velocity
+    all_parameters.manufactured_convergence_study_param.manufactured_solution_param.advection_vector[1] = adv_speed_y; //y-velocity
+
+    int n_time_refinements = 3;
+    double dt_init = 2.5E-3;
+    double dt = dt_init;
+    const double refine_ratio = 0.5;
+    double finalTime = 1.0;
+
+    dealii::ConvergenceTable convergence_table;
+
+    double L2_error_old = 0;
+    double L2_error_conv_rate[4];
+
+    for (int refinement = 0; refinement < n_time_refinements+1; ++refinement){
```

Comment:
In `refinement < n_time_refinements+1` , the +1 is probably not required. Setting `n_time_refinements` to 4 should work too.

Reply:
You're right, I was doing this in a confusing way - I was viewing only calculations after the first ones as "refinement". I'll follow your suggestion so that it is less confusing.